cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlClientTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly.cruisecontrol;
+
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+@ExtendWith(VertxExtension.class)
+public class CruiseControlClientTest {
+
+    private static final int PORT = 1080;
+    private static final String HOST = "localhost";
+
+    private static ClientAndServer ccServer;
+
+    @BeforeAll
+    public static void setupServer() throws IOException, URISyntaxException {
+        ccServer = MockCruiseControl.getCCServer(PORT);
+    }
+
+    @AfterAll
+    public static void stopServer() {
+        ccServer.stop();
+    }
+
+    @Test
+    public void testMockCCServer(Vertx vertx, VertxTestContext context) {
+
+        Promise<String> result = Promise.promise();
+
+        vertx.createHttpClient().get(PORT, HOST, "/test", response -> {
+            response.bodyHandler(body -> {
+                result.complete(body.toString());
+            });
+        }).end();
+
+        result.future().setHandler(bodyFuture -> {
+            if (bodyFuture.succeeded()) {
+                context.verify(() -> assertThat(bodyFuture.result(), is("Was this what you wanted?")));
```

Comment:
Would it be better if the String  "Was this what you wanted?" was kept in a variable in the MockCruiseControl class and then referenced here?

Reply:
We had this discussion more times on the Strimzi project as well. Using a variable could make sense if you need the same text string more times imho (which seems not to be this case).